include/inexor/vulkan-renderer/tools/cla_parser.hpp

```diff
@@ -1,123 +1,100 @@
 ﻿#pragma once
 
-#include <spdlog/spdlog.h>
-
+#include <cstdint>
 #include <optional>
 #include <string>
 #include <unordered_map>
+#include <utility>
 #include <vector>
 
 namespace inexor::vulkan_renderer::tools {
 
-/// @brief Defines the type of an accepted command line argument.
-enum class CommandLineArgumentType {
-    NONE,
-    STRING,
-    INT64,
-    UINT32,
-    BOOL,
-    // TODO: Add more
-};
+/// @brief A command line argument template class
+class CommandLineArgumentTemplate {
+    const std::string name;
+    const bool takes_args;
 
-/// @brief A command line argument template class.
-/// @noe This class does not contain any data yet!
-struct CommandLineArgumentTemplate {
-    CommandLineArgumentTemplate(const CommandLineArgumentType param_type, std::string param_name);
+public:
+    CommandLineArgumentTemplate(std::string name, bool takes_args) : name(std::move(name)), takes_args(takes_args) {}
 
-    std::string argument_name;
+    [[nodiscard]] const std::string &get_name() const {
+        return name;
+    }
 
-    CommandLineArgumentType argument_type;
+    [[nodiscard]] bool does_take_args() const {
+        return takes_args;
+    }
 };
 
-/// @brief Contains the actual command line data.
-struct CommandLineArgumentValue {
-    CommandLineArgumentType type;
+/// @brief Contains the actual command line data
+class CommandLineArgumentValue {
+    const std::string value;
 
-    std::string value_str;
-    std::int64_t value_int64;
-    std::uint32_t value_uint32;
-    bool value_bool;
+public:
+    explicit CommandLineArgumentValue(std::string value) : value(std::move(value)) {}
+
+    template <typename T>
+    T as() const;
 };
 
 /// @brief A simple command line argument parser.
-/// @todo What if an argumen gets specified twice?
+/// @todo What if an argument gets specified twice?
 class CommandLineArgumentParser {
```

Comment:
Add more documentation, also mention that only arugments with a maximum of one value are possible (not splitted by a space).